VENDORING.md

```diff
@@ -0,0 +1,238 @@
+# Perform the vendoring of the Kata Containers project
+
+* [Vendoring tool](#vendoring-tool)
+    * [How to install dep](#how-to-install-dep)
+    * [Overview of the main commands](#overview-of-the-main-commands)
+* [Vendoring use cases](#vendoring-use-cases)
+    * [Initialize vendoring of the repository](#initialize-vendoring-of-the-repository)
+    * [Vendor a newly imported package](#vendor-a-newly-imported-package)
+    * [Update a vendored package](#update-a-vendored-package)
+    * [Remove a vendored package](#remove-a-vendored-package)
```

Comment:
Can you slot in a section here for testing the newly vendored branch? The scenario I'm thinking about is where the user does a successful vendor but forgets to `git add Gopkg.* vendor/` and then pushes their code and raises a PR (which then fails).

We could explain they should:

- `git add` the items mentioned.
- Clean their branch (maybe after taking a backup copy "just in case"):
  ```
  $ git clean -dfx
  ```
- Test the branch.
- *Then* push the branch and raise the PR.

Reply:
Well I am not sure we need to put this here. The developer will vendor, and locally he will be able to check that everything is working properly because of the proper vendoring. But in case he would forget to `git add` the changes brought by the vendoring, the CI would not pass. So here we would be in a case where the local testing would work but not the CI because he didn't do a `git add`. I think that's a bit too much and I assume the developer knows he has to push all the files, even the ones vendored.

Reply:
Ok. I appreciate that it *is* the developers responsibility to do this. But it's easy to forget! However, as you say, the CI is the safety net and warning to the developer they forgot something.